src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -251,9 +270,73 @@ module.exports = function<T, P, I, TI, C, CX>(
     return workInProgress.child;
   }
 
+  function updateHostRoot(current, workInProgress, priorityLevel) {
+    const root = (workInProgress.stateNode : FiberRoot);
+    if (root.pendingContext) {
+      pushTopLevelContextObject(
+        root.pendingContext,
+        root.pendingContext !== root.context
+      );
+    } else {
+      pushTopLevelContextObject(root.context, false);
+    }
+
+    pushHostContainer(root.containerInfo);
+
+    const updateQueue = workInProgress.updateQueue;
+    if (updateQueue) {
+      const prevState = workInProgress.memoizedState;
+      const state = beginUpdateQueue(workInProgress, updateQueue, null, prevState, null, priorityLevel);
+      if (prevState === state) {
+        // If the state is the same as before, that's a bailout because we had
+        // no work matching this priority.
+        return bailoutOnAlreadyFinishedWork(current, workInProgress);
+      }
+      const element = state.element;
+      reconcileChildren(current, workInProgress, element);
+      workInProgress.memoizedState = state;
```

Comment:
GitHub suppressed [my comment on an older commit](https://github.com/facebook/react/pull/8613#discussion_r93501718) so here it is again:

Shouldn't we return `workInProgress.child` inside this branch? I get that `bailoutOnAlreadyFinishedWork` does that anyway but if there's state then it's not really a bailout, is it?